paasta_tools/kubernetes/application/controller_wrappers.py

```diff
@@ -204,15 +204,48 @@ def deep_delete_and_create(self, kube_client: KubeClient) -> None:
             f"Deep Delete {self.kube_deployment.service} {self.kube_deployment.instance}."
         )
         self.deep_delete(kube_client)
+        self.logging.debug(
+            f"waiting for instance {self.kube_deployment.service} {self.kube_deployment.instance} to die."
+        )
+        timer = 0
         while self.kube_deployment in set(list_all_deployments(kube_client)):
-            self.logging.debug(
-                f"waiting for instance {self.kube_deployment.service} {self.kube_deployment.instance} to die."
-            )
+
             sleep(0.5)
+            timer += 0.5
+            if timer >= 3600:
+                try:
+                    self.logging.debug(
+                        f"Force kill {self.kube_deployment.service} {self.kube_deployment.instance}."
+                    )
+                    kube_client.core.delete_namespace(
```

Comment:
delete_namespace? You probably want to list all pods and delete them here. You write some helper functions in kuberntes_tools.py.  Looks good overall otherwise.

Reply:
Im trying this but I feel like the grace_preiod being 0 isnt being respected, I dont see my changes:

Should I add a wait time for 2 seconds in case we  have the pre_stop set? ( Im pretty sure I wait for longer than that tho :|)

```
                all_pods = pods_for_service_instance(service_name, instance_name, kube_client)
                pods_to_delete = filter_pods_by_service_instance(all_pods, service_name , instance_name)
                delete_options = V1DeleteOptions()
                delete_options.grace_period_seconds = 0

                for pod in pods_to_delete:
                    import pdb; pdb.set_trace()
                    response = kube_client.core.delete_namespaced_pod(pod.metadata.name,self.item.metadata.namespace,body=delete_options,grace_period_seconds=0)
```
response looks like this

https://fluffy.yelpcorp.com/i/wjz1ctGN6fNm7dKmSpzJMsDt5nD3HMBW.html